### PR TITLE
feat: add migration script to encrypt old credentials

### PR DESCRIPTION
**Pull request by djabarovgeorge**

### CODE REVIEW

File: apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    const encryptIntegration = await integrationRepository.find({});
+
+    for (const integrationKey in encryptIntegration) {
+      const integration = encryptIntegration[integrationKey];
+
+      expect(integration.credentials.user).to.equal('smith' + integrationKey);
+      expect(integration.credentials.apiKey).to.contains('nvsk.');
+      expect(integration.credentials.secretKey).to.contains('nvsk.');
+    }
```
Comment (scopsy) [inline apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts:50]: Could you add an assertion for encryptIntegration? In cases it's empty the test will always pass :)

File: apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      active: false,
+    };
+
+    for (let i = 0; i < 2; i++) {
+      await integrationRepository.create({
+        _environmentId: session.environment._id,
+        _organizationId: session.organization._id,
+        providerId: data.providerId + i,
+        channel: data.channel,
+        credentials: { user: 'smith' + i, apiKey: '123' + i, secretKey: 'abc' + i },
```
Comment (scopsy) [inline apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts:31]: Maybe worth adding here other secret fields?

**Review by scopsy — APPROVED** [review]
Great. Loved the tests too!

File: apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts (hunk 2 above)
Comment (djabarovgeorge) [reply apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts:31]: Good idea :)

File: apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts (hunk 1 above)
Comment (djabarovgeorge) [reply apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts:50]: Hey @scopsy could you elaborate a bit more not sure I followed

File: apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts (hunk 2 above)
Comment (djabarovgeorge) [reply apps/api/migrations/encrypt-credentials/encrypt-credentials-migration.spec.ts:31]: I asked on another comment :) , 
does that mean that the other comment addresses the same suggestion as here?

File: libs/shared/src/consts/providers/credentials/secure-credentials.ts [hunk 3]
```diff
@@ -4,6 +4,5 @@ export const secureCredentials: CredentialsKeyEnum[] = [
   CredentialsKeyEnum.ApiKey,
   CredentialsKeyEnum.SecretKey,
   CredentialsKeyEnum.Token,
-  CredentialsKeyEnum.Secure,
```
Comment (djabarovgeorge) [inline libs/shared/src/consts/providers/credentials/secure-credentials.ts:7]: @scopsy Do you know of a possible reason why we would hide/encrypt this variable? (i think it is mistaken to be there in the first place) This is a variable that is boolean and at the moment used only on Nodemailer.

Comment (scopsy) [reply libs/shared/src/consts/providers/credentials/secure-credentials.ts:7]: I aggree it shouldn't be secure, even tho the name is a bit confusing 😅

**Pull request merged**